src/sagemaker/inference_recommender/inference_recommender_mixin.py

```diff
@@ -149,12 +148,26 @@ def right_size(
 
         self._init_sagemaker_session_if_does_not_exist()
 
+        model_name = None
+        if isinstance(self, sagemaker.model.FrameworkModel):
+
+            unique_tail = uuid.uuid4()
+            model_name = "SMPYTHONSDK-" + str(unique_tail)
+
+            self.sagemaker_session.create_model(
+                name=model_name,
+                role=self.role,
+                container_defs=None,
+                primary_container=self.prepare_container_def(),
+            )
+
         ret_name = self.sagemaker_session.create_inference_recommendations_job(
             role=self.role,
             job_name=job_name,
             job_type=job_type,
             job_duration_in_seconds=job_duration_in_seconds,
-            model_package_version_arn=self.model_package_arn,
+            model_name=model_name,
+            model_package_version_arn=getattr(self, "model_package_arn", None),
```

Comment:
So whether `self.name` exists or not doesn't matter, because as long as it is a Model object and not a ModelPackage object, we will create our own `temp_model_name` and use that to create the resources and run IR job.
If customer calls `model_package = model.register()` and the `model_package.right_size()` then only the `model_package_arn` will be set, while `self.name` is None. In this case the `temp_model_name` is also None because the object is `sagemaker.model.ModelPackage`

There is one case where `self.model_package_arn` and `self.name` both exist: if customer explicitly do:
```
model_package.name = "some name"
```
We can add validation to prevent it too, thanks!